sound/soc/sof/pm.c

```diff
@@ -422,3 +438,41 @@ int snd_sof_suspend(struct device *dev)
 	return sof_suspend(dev, false);
 }
 EXPORT_SYMBOL(snd_sof_suspend);
+
+int snd_sof_set_dsp_power_state(struct snd_sof_dev *sdev,
+				enum sof_d0_substate d0_substate)
+{
+	u32 flags;
+	int ret;
+
+	dev_dbg(sdev->dev, "setting ADSP D0 substate:%d\n", d0_substate);
+
+	/* Todo: destroy/reload unused pipelines/widgets in FW */
+
+	/* do platform specific set_state */
+	ret = snd_sof_dsp_set_power_state(sdev, d0_substate);
+	if (ret < 0) {
+		dev_err(sdev->dev,
+			"error: d0i3_exit ipc error %d\n", ret);
+		return ret;
+	}
+
+	if (d0_substate == SOF_DSP_D0I3)
+		flags = SOF_PM_PG_STREAMING;
+	else
+		flags = SOF_PM_PPG | SOF_PM_PCG;
```

Comment:
I really don't think this.
The flags should be added in the platform-specific parts, not in the core
How is this even remotely portable across architectures? Even across Intel solutions, this will not scale since I bet in some cases we need to have quirks.

Reply:
we can send this to FW(as our requirement), and FW will decide how it should handle it in different platforms?

Reply:
those are Intel specific things, why export them and make them visible to the core?

Reply:
OK, then a thought pops in my head, lets treat the flag vendor/platform specific, and fill in it in platform's set_power_state() ops.